mm/ktsan/report.c

```diff
@@ -149,12 +148,20 @@ void kt_report_race(kt_thr_t *new, kt_race_info_t *info)
 			(1 << info->old.size), info->old.tid);
 		pr_err("No stack available.\n");
 	} else {
-		pr_err("Previous %s of size %d by thread T%d (K%d):\n",
+		cpu_id = kt_trace_restore_cpu_id(old, info->old.clock);
```

Comment:
kt_trace_restore_cpu_id should be combined with kt_trace_restore_stack.
That is, we walk trace once and extract all necessary info from trace (stack, cpu, mutexset, etc).
Believe it or not, just yesterday I've optimized stack restoration in user-space tsan because it was a performance bottleneck.